### ISSUE TITLE
Alias in delete makes invalid query

### ISSUE THREAD

**Issue by puggan** [laravel/framework#18085]
- Laravel Version: 5.1.40
- PHP Version: 5.6.15
- Database Driver & Version: 10.1.21-MariaDB

### Description:
When trying to remove duplicates before adding a unique key in a migration, I got the error:

> [Illuminate\Database\QueryException]
> SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'as `usi_high` from `user_seller_info` as `usi_high` inner join `user_seller_info' at line 1 (SQL: delete `user_seller_info` as `usi_high` from `user_seller_info` as `usi_high` inner join `user_seller_info` as `usi_low` on `usi_low`.`user_id` = `usi_high`.`user_id` and `usi_low`.`id` < `usi_high`.`id`)

### Steps To Reproduce:
1. starta query-builder, using an alias on the table name
2. do a join (maybe not a required step)
3. delete()

Example:
```
        DB::table('user_seller_info AS usi_high')
            ->join(
                'user_seller_info AS usi_low',
                function ($join) {
                    $join
                        ->on('usi_low.user_id', '=', 'usi_high.user_id')
                        ->on('usi_low.id', '<', 'usi_high.id');
                }
            )
            ->delete();
```

### Current result:
```delete `user_seller_info` as `usi_high` from ... ```

### Expected result:
```delete `usi_high` from ... ```

**Comment by Dylan-DPC** [comment]
@puggan You can close this issue.

**Comment by puggan** [comment]
It's fixed in 5.5, 28/2.
https://github.com/laravel/framework/commit/d2b6c8bc5cc443e41bcfe344973e105b728bebb5

But the 5.1 dosn't seams to be fixed,
https://github.com/laravel/framework/blob/5.1/src/Illuminate/Database/Query/Grammars/MySqlGrammar.php

But as the 5.1-LTS died this year, I guess it's not going to be solved.

**Comment by Dylan-DPC** [comment]
yeah you are spot on. 5.1 is EOL so it won't be fixed (unless it is a security issue i bet)
